File: leetcode/validAnagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    
    if(s.length !== t.length) return false;
    
    const trackerS = {};
    const trackerT = {};
    
    for(let letter of s){
        if(trackerS[letter]){
            trackerS[letter] += 1;
        }else{
            trackerS[letter] = 1;
        }
    }
    
    for(let letter of t){
        if(trackerT[letter]){
            trackerT[letter] += 1;
        }else{
            trackerT[letter] = 1;
        }
    }
    
    for(let key in trackerS){
        if(!trackerT[key]) return false;
        
        if(trackerT[key] !== trackerS[key]){
            return false;
        }
    }
    
    return true;
};

File: leetcode/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        trackerS = {}
        trackerT = {}
        
        for letter in s:
            if letter in trackerS:
                trackerS[letter] += 1
            else:
                trackerS[letter] = 1
        
        for letter in t:
            if letter in trackerT:
                trackerT[letter] += 1
            else:
                trackerT[letter] = 1
        
        for key in trackerS:
            if key in trackerT:
                if trackerS[key] == trackerT[key]:
                    pass
                else:
                    return False
            else:
                return False
        
        return True
